pkg/cmd/cli/backup/create.go

```diff
@@ -107,7 +109,7 @@ type CreateOptions struct {
 	CSISnapshotTimeout              time.Duration
 	ItemOperationTimeout            time.Duration
 	ResPoliciesConfigmap            string
-	client                          veleroclient.Interface
+	client                          dynamic.Interface
```

Comment:
Seems we have both the dynamic client and controller-runtime client in our code, could we just keep one of them? How about to use the controller-runtime client which is easy to use?

Reply:
The backup and restore create sub-command has a --wait option. Code uses an informer to watch the backup's change. The controller-runtime framework doesn't encourage users to use Informer with it. This is the main reason for choosing the dynamic client.

Reply:
Just curious why `the controller-runtime framework doesn't encourage users to use Informer with it`?

Is it for the client provided by the controller manager which embeds informers? But for client we used in the CLI isn't the one provided by the controller manager, it has no informers embeded.

Reply:
https://github.com/kubernetes-sigs/controller-runtime/issues/1668
I found this issue, and the comments there made me feel it's not encouraged.
I agree that passing an informer not working doesn't mean the internal informer cannot be configured.
If using a customized source, and writing the data into a channel when the condition matches, it should also work.

The only downside is we need to write more boilerplate code for controller-runtime, such as manager and reconcile.

Reply:
Removed the dynamic client.